flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -309,6 +312,25 @@ public TaskExecutor(
 		return CompletableFuture.completedFuture(shuffleEnvironment.getPartitionsOccupyingLocalResources().isEmpty());
 	}
 
+	@Override
+	public CompletableFuture<Collection<LogInfo>> requestLogList(Time timeout) {
+		final String logDir = taskManagerConfiguration.getTaskManagerLogDir();
+		if (logDir != null) {
+			final File[] logFiles = new File(logDir).listFiles();
+
+			if (logFiles == null) {
+				return FutureUtils.completedExceptionally(new FlinkException(String.format("There isn't a log file in TaskExecutor’s log dir %s.", logDir)));
+			}
+
+			final List<LogInfo> logsWithLength = Arrays.stream(logFiles).filter(File::isFile)
+				.map(logFile -> new LogInfo(logFile.getName(), logFile.length()))
+				.collect(Collectors.toList());
+			return CompletableFuture.completedFuture(logsWithLength);
+		}
+		//There isn't log file available on the TaskExecutor.
```

Comment:
This comment is superfluous. I would remove it. The method isn't very long and also from the exception text it should be obvious.